rpcs3/Emu/Memory/vm_locking.h

```diff
@@ -50,50 +50,22 @@ namespace vm
 		// Old-style conditional constexpr
 		const u32 size = Size ? Size : _size;
 
-		if (size <= 4096u && !((begin | size) & (size - 1)) ? !vm::check_addr(begin) : !vm::check_addr(begin, vm::page_readable, size))
+		if (Size == 1 || (begin % 4096 + size % 4096) / 4096 == 0 ? !vm::check_addr(begin) : !vm::check_addr(begin, vm::page_readable, size))
 		{
 			range_lock->release(0);
 			range_lock_internal(range_lock, begin, _size);
 			return;
 		}
 
-		const u64 lock_val = g_range_lock.load();
-		const u64 is_share = g_shmem[begin >> 16].load();
-
 		#ifndef _MSC_VER
 		__asm__(""); // Tiny barrier
 		#endif
 
-		u64 lock_addr = static_cast<u32>(lock_val); // -> u64
-		u32 lock_size = static_cast<u32>(lock_val << range_bits >> (32 + range_bits));
-
-		u64 addr = begin;
-
-		// Optimization: if range_locked is not used, the addr check will always pass
```

Comment:
Where did this code go?

Reply:
The complete check is now in the function so register limit won't be reached, most of the time  it's unlocked so the initial check suffices. I've optimized the underlying function for it.